src/optimizer/core/src/collector.rs

```diff
@@ -434,9 +434,34 @@ impl Visit for HookCollect {
         self.expr_ctxt.pop();
     }
 
+    fn visit_jsx_opening_element(&mut self, node: &ast::JSXOpeningElement) {
+        let mut stacked = false;
+        if let ast::JSXElementName::Ident(ref ident) = node.name {
+            let ident_name = ident.sym.as_ref().chars().next().unwrap();
+            if let 'A'..='Z' = ident_name {
```

Comment:
Maybe you can avoid the `unwrap` by also pattern matching on the `Option`
```suggestion
            let ident_name = ident.sym.as_ref().chars().next();
            if let Some('A'..='Z') = ident_name {
```